static/src/javascripts/projects/common/modules/identity/consent-journey.js

```diff
@@ -58,11 +73,28 @@ const showJourneyAlert = (journeyEl: HTMLElement): void => {
         formEl.addEventListener('submit', ev => {
             if (ev.isTrusted) {
                 ev.preventDefault();
-                Promise.all([
-                    shouldDisplayNewsletterWarning(journeyEl),
-                    shouldDisplayConsentWarning(journeyEl),
-                ]).then(([emptyNewsletters, emptyMarketing]) => {
-                    if (emptyNewsletters && emptyMarketing) {
+                const uncheckBoxesPromise = Promise.all([
+                    uncheckedBoxes(journeyEl, 'email'),
+                    uncheckedBoxes(journeyEl, 'marketing-consents'),
+                ]).then(checkboxes =>
+                    [].concat(...checkboxes).filter(_ => _.checked === false)
+                );
+                uncheckBoxesPromise.then(unchecked => {
+                    if (unchecked.length > 0) {
+                        const consentNames = unchecked.map(
+                            uncheckedbox =>
+                                uncheckedbox.parentElement.querySelector(
+                                    '.manage-account__switch-title'
+                                ).innerText
+                        );
+                        getContents('confirm-consents').then(modalEl => {
+                            if (consentNames) {
```

Comment:
this is always gonna be true, isn't it?

Reply:
i guess you _could_ filter consentNames for empty names first then run the check, to let cases where the dom is messed up fail silently but that should happen higer down the line so the form submits? i don't know it's not really a huge deal

Reply:
yep supposed to be consentNames.length > 0, updated